x-pack/plugins/gis/public/shared/components/layer_toc_actions.js

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component } from 'react';
+
+import {
+  EuiButtonEmpty,
+  EuiPopover,
+  EuiContextMenu,
+  EuiIcon,
+  EuiLoadingSpinner,
+  EuiToolTip,
+  EuiIconTip
+} from '@elastic/eui';
+
+
+function flattenPanelTree(tree, array = []) {
+  array.push(tree);
+
+  if (tree.items) {
+    tree.items.forEach(item => {
+      if (item.panel) {
+        flattenPanelTree(item.panel, array);
+        item.panel = item.panel.id;
+      }
+    });
+  }
+
+  return array;
+}
+
+export class LayerTocActions extends Component {
+
+
+  constructor() {
+
+    super();
+    this.state = {
+      isPopoverOpen: false
+    };
+
+  }
+
+  _onClick = () => {
+    this.setState(prevState => ({
+      isPopoverOpen: !prevState.isPopoverOpen,
+    }));
+  };
+
+  _closePopover = () => {
+    this.setState(() => ({
+      isPopoverOpen: false
+    }));
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
```

Comment:
isMounted is never used. You can just remove `componentDidMount `and `componentWillUnmount ` functions